### PR TITLE
feat(rfq-relayer): add mutex on committable balance consumption

### PR DESCRIPTION
**Pull request by dwasse**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new locking mechanism for balance management to enhance concurrency control.
  - Added a mutex for balance requests in the relayer to synchronize access.

- **Improvements**
  - Enhanced the `QuoteRequestHandler` with a mutex for thread-safe balance operations.
  - Improved documentation for mutex usage in the relayer struct for clarity.

### CODE REVIEW

File: services/rfq/relayer/service/handlers.go
Comment (ChiTimesChi) [inline services/rfq/relayer/service/handlers.go:1]: @dwasse I believe we also need to acquire the balance mutex in `handleNotEnoughInventory`, where we also do `GetCommittableBalance -> assign CommittedPending`

File: services/rfq/relayer/service/handlers.go [hunk 1]
```diff
@@ -162,6 +162,17 @@ func (q *QuoteRequestHandler) handleSeen(ctx context.Context, span trace.Span, r
...
+	// lock the consumed balance
+	key := getBalanceMtxKey(q.Dest.ChainID, request.Transaction.DestToken)
+	span.SetAttributes(attribute.String("balance_lock_key", key))
+	unlocker, ok := q.balanceMtx.TryLock(key)
+	if !ok {
+		// balance is locked due to concurrent request, try again later
+		span.SetAttributes(attribute.Bool("locked", true))
+		return nil
+	}
+	defer unlocker.Unlock()
```
Comment (ChiTimesChi) [inline services/rfq/relayer/service/handlers.go:174]: In the case of forwarding, won't the lock be released only after the full cycle goes through? We could safely unlock once we update status to `CommittedPending` (aka spend the destination commitable balance)

Comment (ChiTimesChi) [reply services/rfq/relayer/service/handlers.go:174]: @dwasse by "full cycle" I mean the full status path `Seen -> CommitPending -> CommittedConfirmed`. I realised that it's not actually full, but still if that's the case we're keeping the lock longer than we should've

Comment (dwasse) [reply services/rfq/relayer/service/handlers.go:174]: Yeah that makes sense, would still like to use `defer` to make this error resistant but can just add a helper function here

**Pull request merged**